lib/charms/hydra/v0/oauth.py

```diff
@@ -662,6 +694,17 @@ def _delete_juju_secret(self, relation: Relation) -> None:
         secret = self.model.get_secret(label=self._get_secret_label(relation))
         secret.remove_all_revisions()
 
+    def _pop_client_data(self, relation_id: Relation) -> None:
+        if len(self.model.relations) == 0:
+            return None
+
+        relation = self.model.get_relation(self._relation_name, relation_id=relation_id)
+        if not relation or not relation.app:
+            return None
```

Comment:
The convention is to not specify the None if the return type is `None` (ie not optional)
```suggestion
            return
```